mlir/tools/rocmlir-gen/rocmlir-gen.cpp

```diff
@@ -1268,6 +1267,152 @@ getConv2dBounds(rock::ConvOpType dir,
   return std::make_tuple(dim, dimH, dimW);
 }
 
+static func::FuncOp getMemcpyFuncDecl(ModuleOp module, const MemRefType srcType,
```

Comment:
This function and emitMemcpy() moved up, unchanged, to be available for ensureFloatIsF32().